jetty-core/jetty-server/src/main/java/org/eclipse/jetty/server/handler/DelayedHandler.java

```diff
@@ -39,29 +47,79 @@ public boolean process(Request request, Response response, Callback callback) th
         if (next == null)
             return false;
 
-        DelayedProcess delayed = newDelayedProcess(next, request, response, callback);
+        boolean contentExpected = false;
+        String contentType = null;
+        loop: for (HttpField field : request.getHeaders())
+        {
+            HttpHeader header = field.getHeader();
+            if (header == null)
+                continue;
+            switch (header)
+            {
+                case CONTENT_TYPE:
+                    contentType = field.getValue();
+                    break;
+
+                case CONTENT_LENGTH:
+                    contentExpected = field.getLongValue() > 0;
+                    break;
+
+                case TRANSFER_ENCODING:
+                    contentExpected = field.contains(HttpHeaderValue.CHUNKED.asString());
+                    break;
+
+                case EXPECT:
+                    if (field.contains(HttpHeaderValue.CONTINUE.asString()))
+                    {
+                        contentExpected = false;
+                        break loop;
+                    }
+                    break;
+                default:
+                    break;
+            }
+        }
+
+        MimeTypes.Type mimeType = MimeTypes.getBaseType(contentType);
+        DelayedProcess delayed = newDelayedProcess(contentExpected, contentType, mimeType, next, request, response, callback);
         if (delayed == null)
             return next.process(request, response, callback);
 
-        delay(delayed);
+        delayed.delay();
         return true;
     }
 
-    protected DelayedProcess newDelayedProcess(Handler next, Request request, Response response, Callback callback)
+    protected DelayedProcess newDelayedProcess(boolean contentExpected, String contentType, MimeTypes.Type mimeType, Handler handler, Request request, Response response, Callback callback)
```

Comment:
This looks quite complex to override, this contract should be simplified.

Reply:
Definitely.   Proposals?